modules/advanced_search/js_modules/route_handlers.js

```diff
@@ -62,4 +62,77 @@ function applyAdvancedSearchPageHandlers() {
         }
     }
     Hm_Message_List.check_empty_list();
+
+    $('body').on("click", ".pick_special_folders", function(e) {
+        e.preventDefault();
+
+        const modal = new Hm_Modal({
+            modalId: 'pick_special_folders_modal',
+            title: 'Pick Special Folders',
+            size: 'lg',
+        });
+
+        const serverId = $(this).closest('li').attr("class").split('_')[1];
+        const specialFolders = [];
+        ['archive', 'draft', 'junk', 'sent', 'trash'].forEach(folderType => {
+            const folderName = hm_special_folders()[serverId][folderType];
+            if (folderName) {
+                specialFolders.push(folderName);
+            }
+        });
+
+        const account = $(this).closest('li').find('.adv_folder_link');
+        const accountLabel = account.text();
+        const accountId = account.data('target');
+
+        const isFolderSelected = (folder) => get_adv_sources().find(source => source.label === getFolderLabel(folder));
+
+        const getFolderLabel = (folder) => accountLabel + ' > ' + folder;
+
+        modal.setContent(`
+        <div class="d-flex gap-3 flex-wrap">
+            <div class="form-check form-switch">
+                <input class="form-check-input" type="checkbox" id="all">
```

Comment:
Please use a bit more specific identifier here - `all` might conflict with another one in current or future version of Cypht or Tiki.